xarray/core/accessors.py

```diff
@@ -0,0 +1,139 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from .common import is_datetime_like
+from .pycompat import dask_array_type
+
+import numpy as np
+import pandas as pd
+
+
+def _season_from_months(months):
+    """Compute season (DJF, MAM, JJA, SON) from month ordinal
+    """
+    # TODO: Move "season" accessor upstream into pandas
+    seasons = np.array(['DJF', 'MAM', 'JJA', 'SON'])
+    months = np.asarray(months)
+    return seasons[(months // 3) % 4]
+
+
+def _access_through_series(values, name):
+    """Coerce an array of datetime-like values to a pandas Series and
+    access requested datetime component
+    """
+    values_as_series = pd.Series(values.ravel())
+    if name == "season":
+        months = values_as_series.dt.month.values
+        field_values = _season_from_months(months)
+    else:
+        field_values = getattr(values_as_series.dt, name).values
+    return field_values.reshape(values.shape)
+
+
+def _get_date_field(values, name):
+    """Indirectly access pandas' libts.get_date_field by wrapping data
+    as a Series and calling through `.dt` attribute.
+
+    Parameters
+    ----------
+    values : np.ndarray or dask.array-like
+        Array-like container of datetime-like values
+    name : str
+        Name of datetime field to access
+
+    Returns
+    -------
+    datetime_fields : same type as values
+        Array-like of datetime fields accessed for each element in values
+
+    """
+    if isinstance(values, dask_array_type):
+        from dask.array import map_blocks
+        out_dtype = object if name == "weekday_name" else np.int64
```

Comment:
Should also be `object` for season and time.

Actually, rather than put the logic here, can you add a `dtype` argument to `_tslib_field_accessor` and pass it on?